xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -736,7 +738,6 @@ MatchFwdResult MatchBmm1ScaleBiasMaskSoftmaxDropoutBmm2(
       matched_result.has_match = false;
       return matched_result;
     }
-    matched_result.is_causal_mask |= IsCausalMaskPattern(mask);
```

Comment:
Why do we need to remove this line?

Reply:
Current CausalMask pattern match only match a causal mask that is added to the bmm1 output. Would be better to do this in a separate PR.

Reply:
Also require to rewrite some parts of rewriter logic to consider cases where bias and mask could both be causalMask.

Reply:
If I understand correctly, not removing this line would currently be an issue and you intend to add some additional logic in another PR? Sounds good.